### ISSUE TITLE
Internal server error on first time connection to Web UI

### ISSUE THREAD

**Issue by mkingsbury-orcadt** [mistio/mist-ce#957]
Brand new installation of mist-ce v4.1.  docker-compose up -d, verified settings/settings.py, valid URI (http, not https), docker-compose restart .  No SSL/TLS certs installed or configured.  

Trying to reach the UI via the URL provided in CORE_URI I get "Internal server error".  

edit - Discovered going to $URL/server got me a 404 error page with a link to login. Navigating there got me into the interface and it appears to be functioning properly from there. 

Traceback in log -

api_1                   | 2019-06-22 02:19:34,450 CRITI [mist.api.views][uWSGIWorker3Core0] Uncaught non-mist exception? WTF!
api_1                   | Traceback (most recent call last):
api_1                   |   File "/mist.api/src/mist/api/views.py", line 163, in home
api_1                   |     user = user_from_request(request)
api_1                   |   File "/mist.api/src/mist/api/auth/methods.py", line 185, in user_from_request
api_1                   |     raise UserUnauthorizedError()
api_1                   | mist.api.exceptions.UserUnauthorizedError: User not authenticated. Please log in

**Comment by d-mo** [comment]
I tried a brand new mise-ce v4.1 install and couldn't reproduce the issue. Does it persist in your case? Could it have been a hick-up due to some service not being ready yet? Are there any more exceptions in the API log? 
Also, please confirm that all services are up with `docker-compose ps`

**Comment by mkingsbury-orcadt** [comment]
Services were all up.  I had an earlier issue with selinux preventing docker-compose from properly bringing up all the services and had disabled selinux, restarted, docker-compose down -v,  before starting over. 

A fresh install on a fresh VM with selinux disabled at the beginning was successful. 
A fresh install on a fresh VM with selinux enabled, failed to bring it up, disabled selinux & restarted, and mist came up successfully. 

API log is attached.  I apologize, there's HTTPConnection errors in there,  I hadn't seen the root mist splash screen that I'm guessing (?) is getting pulled over the internet, wondering if something's slightly off there. 

While I can't reproduce the problem on a new VM, I am curious to discover the problem.

[mist_api.log](https://github.com/mistio/mist-ce/files/3321143/mist_api.log)

**Comment by aljaxus** [comment]
I can confirm the issue. I am also getting the "internal server error" on a just installed CentOS 7 dedi, fresh install on everything.
```
settings.py

"""User defined settings"""

## Configure external url.
CORE_URI = "http://192.168.1.101"

## Verify SSL outgoing SSL connections.
# SSL_VERIFY = True

## Configure email settings.
# MAILER_SETTINGS = {
#     'mail.host': "mailmock",
#     'mail.port': "8025",
#     'mail.tls': False,
#     'mail.starttls': False,
#     'mail.username': "",
#     'mail.password': "",
# }

## Configure whether to serve bundled/minified UI assets.
JS_BUILD = True

## Allow mist.io to connect to localhost.
#ALLOW_CONNECT_LOCALHOST = True

## Allow mist.io to connect to private IP addresses.
#ALLOW_CONNECT_PRIVATE = True

## API token for github.
# GITHUB_BOT_TOKEN = ""
```

**Comment by aljaxus** [comment]
I just tested out with v4.0.3 and it seems to work just fine

**Comment by shawnweeks** [comment]
I can confirm this issue as well. New installation of CentOS 7. Never comes up with SELinux Enabled. Comes up with SELinux Disabled but gives me an internal server error until I goto /server. Then the homepage loads just fine.

**Comment by d-mo** [comment]
Fixed on [v4.1.1](https://github.com/mistio/mist-ce/releases/tag/v4.1.1)
